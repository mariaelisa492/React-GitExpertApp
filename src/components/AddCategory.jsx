import React, {useState} from 'react';
import PropTypes from 'prop-types'


export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        setInputValue(e.target.value)                     //obtengo todo lo que entra en el input
    }

    const handleSubmit = (e) =>{
        e.preventDefault();         //previene que se recargue la pagina con el submit(recibe el evento del enter)
            if (inputValue.trim().length > 2){
                setCategories(cats => [inputValue, ...cats,]);
                setInputValue('');
            }
    }

    return (
        <form onSubmit={handleSubmit}>           
            <input
            type="text"
            placeholder='Buscar'
            value = {inputValue}              //el value se va actualizando
            onChange = {handleInputChange}         //onChange genera los cambios enviadole el evento al setState
            />
        </form>
    )

}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
