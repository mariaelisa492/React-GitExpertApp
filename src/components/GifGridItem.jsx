import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({title, url}) => {
    return (
        <>
        <div className='card animate__animated animate__fadeInDown'>
            <img src={url} alt={title} />
            <h4>{title}</h4>
        </div>
        </>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}