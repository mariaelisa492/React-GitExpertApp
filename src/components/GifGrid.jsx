import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';


export const GifGrid = ({category}) => {
    const {data: images, loading} = useFetchGifs(category);

    return (
        <>
        <h3 className='card animate__animated animate__flipInY animate__delay-1s'>{category}</h3>
        {loading && <p className='card animate__animated animate__flash'>Loading</p>}

        <div className='card-grid'>
            {
                images.map(img => (
                    <GifGridItem 
                    key={img.id}
                    {...img}/>   //aqui estoy mandando cada propiedad de las props como propiedad independiente
                ))
            }
        </div>
        </>
    )
}
