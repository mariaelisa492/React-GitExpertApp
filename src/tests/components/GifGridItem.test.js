import React from "react"
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem />', ()=>{

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);  //yo puse como propTypes a la url y al title

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener un h4 con el titulo', () => {
        const h4 = wrapper.find('h4');
        expect(h4.text().trim()).toBe(title);
        
    })

    test('Debe tener el src y el alt de la img', () => {
        const img = wrapper.find('img');
        // console.log(img.props().src);   o podria ser asi img.prop('src)
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    })

    test('Debe tener el div con el className', () => {
        const div = wrapper.find('div');
        console.log(div.props());
        const className = div.prop('className');
        expect(className.includes('animate__fadeInDown')).toBe(true)  //puedo usar el includes porque es el className es un strgin
    })
})
