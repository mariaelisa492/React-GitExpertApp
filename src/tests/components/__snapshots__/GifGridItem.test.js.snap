// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeInDown"
  >
    <img
      alt="Un titulo"
      src="https://localhost/algo.jpg"
    />
    <h4>
      Un titulo
    </h4>
  </div>
</Fragment>
`;
